### PR TITLE
Add extension.json and a posible fix for composer

### PR DESCRIPTION
**Pull request by paladox**

### CODE REVIEW

**Comment by paladox** [comment]
I'm not sure why it keeps failing but the fix should be something like this maybe not extension.json.

**Comment by paladox** [comment]
Hi could I have help to fix the error.

File: composer.json [hunk 1]
```diff
@@ -32,7 +32,15 @@
 		}
 	},
 	"autoload": {
+		"classmap": [
```
Comment (JeroenDeDauw) [inline composer.json]: Don't autoloade things from other components! They should do this themselves

File: SemanticFormsSelect.hooks.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+class SemanticFormsSelectHooks {
```
Comment (mwjames) [inline SemanticFormsSelect.hooks.php]: Why do I need such class here and especially out of the normal PSR-4 / src directory.

I __did not clean-up__ this extension to see stuff like this (which is common practice in WMF related development) to be floating around.

File: SemanticFormsSelect.php [hunk 3]
```diff
@@ -6,18 +6,27 @@
...
-}
 
-if ( version_compare( $GLOBALS[ 'wgVersion' ], '1.23', 'lt' ) ) {
+if ( version_compare( $GLOBALS[ 'wgVersion' ], '1.23c', 'lt' ) ) {
 	die( '<b>Error:</b> This version of <a href="https://github.com/SemanticMediaWiki/SemanticFormsSelect/">SemanticFormsSelect</a> is only compatible with MediaWiki 1.23 or above. You need to upgrade MediaWiki first.' );
 }
 
-if ( !defined( 'SF_VERSION' ) || version_compare( SF_VERSION, '2.8', 'lt' ) ) {
-   die( '<b>Error:</b> This version of <a href="https://github.com/SemanticMediaWiki/SemanticFormsSelect/">SemanticFormsSelect</a> is only compatible with Semantic Forms 2.8 or above. You need to upgrade <a href="https://www.mediawiki.org/wiki/Extension:Semantic_Forms">Semantic Forms</a> first.' );
+if ( isset( $wgWikimediaTravisCI ) && $wgWikimediaTravisCI == true ) {
```
Comment (mwjames) [inline SemanticFormsSelect.php]: This makes me sad, unrelated code being injected here.

File: SemanticFormsSelect.php [hunk 4]
```diff
@@ -6,18 +6,27 @@
...
+	if ( is_readable( __DIR__ . '/../../vendor/SemanticForms/SemanticForms.php' ) ) {
+		require_once __DIR__ . '/../../vendor/SemanticForms/SemanticForms.php';
```
Comment (mwjames) [inline SemanticFormsSelect.php]: This cannot be right, there is a reason for using Composer to avoid such non-sense here.

File: composer.json [hunk 5]
```diff
@@ -32,7 +32,15 @@
...
+			"vendor/SemanticForms/includes/",
```
Comment (mwjames) [inline composer.json]: I don't think so.

File: extension.json [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+	"url": "https://www.mediawiki.org/wiki/Extension:SemanticFormsSelect",
+	"descriptionmsg": "semanticformsselect-desc",
+	"type": "semantic",
+	"license-name": "GPL-2.0+",
+	"callback": "SemanticFormsSelectHooks::onCallback",
+	"ExtensionFunctions": [
+		"SemanticFormsSelectHooks::onExtensionFunction"
+	],
+	"AutoloadClasses": {
+		"SemanticFormsSelectHooks": "SemanticFormsSelect.hooks.php"
```
Comment (mwjames) [inline extension.json]: Autoloading is done by Composer, no way manual class registration starts to creep in.

**Comment by mwjames** [comment]
After just a quick look, I have to say that my opposition on the whole `extension.json` stuff is growing.

File: SemanticFormsSelect.php (hunk 4 above)
Comment (paladox) [inline SemanticFormsSelect.php]: Since mediawiki core merges the packages from extensions into its own vendor folder so when travis tests the extension the vendor folder wont be created in the extension it self it will be created in mediawiki core. So for a travis specific hack lets do this. And since semanticforms is not composer ready meaning it is avalible in composer but wont be able to be loaded through composer so there wont be any psr code or anything.

File: SemanticFormsSelect.hooks.php (hunk 2 above)
Comment (paladox) [inline SemanticFormsSelect.hooks.php]: So that we can use a callback in extension.json and ExtensionFunctions

**Comment by paladox** [comment]
https://github.com/SemanticMediaWiki/SemanticFormsSelect/pull/20
